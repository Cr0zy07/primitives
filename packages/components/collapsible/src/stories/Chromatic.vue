<script setup lang="ts">
import { OkuCollapsible, OkuCollapsibleContent, OkuCollapsibleTrigger } from '@oku-ui/collapsible'
</script>

<template>
  <h1>Uncontrolled</h1>
  <h2>Closed</h2>
  <OkuCollapsible class="collapsible">
    <OkuCollapsibleTrigger class="collapsible-trigger">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-content">
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>

  <h2>Open</h2>
  <OkuCollapsible class="collapsible" default-open>
    <OkuCollapsibleTrigger class="collapsible-trigger">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-content">
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>

  <h1>Controlled</h1>
  <h2>Closed</h2>
  <OkuCollapsible class="collapsible" :open="false">
    <OkuCollapsibleTrigger class="collapsible-trigger">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-content">
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>

  <h2>Open</h2>
  <OkuCollapsible class="collapsible" open>
    <OkuCollapsibleTrigger class="collapsible-trigger">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-content">
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>

  <h1>Disabled</h1>
  <OkuCollapsible class="collapsible" disabled>
    <OkuCollapsibleTrigger class="collapsible-trigger">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-content">
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>

  <h1>State attributes</h1>
  <h2>Closed</h2>
  <OkuCollapsible class="collapsible-attr-styles">
    <OkuCollapsibleTrigger class="collapsible-attr-styles">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent
      class="collapsible-attr-styles"
      :style="{
        // ensure we can see the content (because it has `hidden` attribute)
        display: 'block',
      }"
    >
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>

  <h2>Open</h2>
  <OkuCollapsible class="collapsible-attr-styles" default-open>
    <OkuCollapsibleTrigger class="collapsible-attr-styles">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent
      class="collapsible-attr-styles"
      :style="{
        // ensure we can see the content (because it has `hidden` attribute)
        display: 'block',
      }"
    >
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>

  <h2>Disabled</h2>
  <OkuCollapsible class="collapsible-attr-styles" default-open disabled>
    <OkuCollapsibleTrigger class="collapsible-attr-styles">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent
      class="collapsible-attr-styles"
      :style="{
        // ensure we can see the content (because it has `hidden` attribute)
        display: 'block',
      }"
    >
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>
</template>
