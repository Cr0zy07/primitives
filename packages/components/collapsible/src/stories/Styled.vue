<script setup lang="ts">
import { OkuCollapsible, OkuCollapsibleContent, OkuCollapsibleTrigger } from '@oku-ui/collapsible'
</script>

<template>
  <OkuCollapsible class="collapsible">
    <OkuCollapsibleTrigger class="collapsible-trigger">
      Trigger
    </OkuCollapsibleTrigger>
    <OkuCollapsibleContent class="collapsible-content">
      Content 1
    </OkuCollapsibleContent>
  </OkuCollapsible>
</template>
