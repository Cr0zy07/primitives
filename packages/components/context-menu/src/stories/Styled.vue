<script setup lang="ts">
import { OkuContextMenu, OkuContextMenuContent, OkuContextMenuItem, OkuContextMenuPortal, OkuContextMenuSeparator, OkuContextMenuTrigger } from '@oku-ui/context-menu'
</script>

<template>
  <div
    :style="{
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      width: '200vw',
      height: '200vh',
      gap: '20px',
    }"
  >
    <OkuContextMenu>
      <OkuContextMenuTrigger class="context-menu-trigger">
        Right click here
      </OkuContextMenuTrigger>
      <OkuContextMenuPortal>
        <OkuContextMenuContent class="context-menu-content" :align-offset="-5">
          <OkuContextMenuItem class="context-menu-item" @select="console.log('undo')">
            Undo
          </OkuContextMenuItem>
          <OkuContextMenuItem class="context-menu-item" @select="console.log('redo')">
            Redo
          </OkuContextMenuItem>
          <OkuContextMenuSeparator class="context-menu-separator" />
          <OkuContextMenuItem class="context-menu-item" disabled @select="console.log('cut')">
            Cut
          </OkuContextMenuItem>
          <OkuContextMenuItem class="context-menu-item" @select="console.log('copy')">
            Copy
          </OkuContextMenuItem>
          <OkuContextMenuItem class="context-menu-item" @select="console.log('paste')">
            Paste
          </OkuContextMenuItem>
        </OkuContextMenuContent>
      </OkuContextMenuPortal>
    </OkuContextMenu>
  </div>
</template>
