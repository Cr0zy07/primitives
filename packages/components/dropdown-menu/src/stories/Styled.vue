<script setup lang="ts">
import { OkuDropdownMenu, OkuDropdownMenuArrow, OkuDropdownMenuContent, OkuDropdownMenuItem, OkuDropdownMenuPortal, OkuDropdownMenuSeparator, OkuDropdownMenuTrigger } from '@oku-ui/dropdown-menu'
</script>

<template>
  <div :style="{ display: 'flex', alignItems: 'center', justifyContent: 'center', height: '200vh' }">
    <OkuDropdownMenu>
      <OkuDropdownMenuTrigger class="dropdown-menu-trigger">
        Open
      </OkuDropdownMenuTrigger>
      <OkuDropdownMenuPortal>
        <OkuDropdownMenuContent class="dropdown-menu-content" :side-offset="5">
          <OkuDropdownMenuItem class="dropdown-menu-item" @select="console.log('undo')">
            Undo
          </OkuDropdownMenuItem>
          <OkuDropdownMenuItem class="dropdown-menu-item" @select="console.log('redo')">
            Redo
          </OkuDropdownMenuItem>
          <OkuDropdownMenuSeparator class="dropdown-menu-separator" />
          <OkuDropdownMenuItem class="dropdown-menu-item" disabled @select="console.log('cut')">
            Cut
          </OkuDropdownMenuItem>
          <OkuDropdownMenuItem class="dropdown-menu-item" @select="console.log('copy')">
            Copy
          </OkuDropdownMenuItem>
          <OkuDropdownMenuItem class="dropdown-menu-item" @select="console.log('paste')">
            Paste
          </OkuDropdownMenuItem>
          <OkuDropdownMenuArrow />
        </OkuDropdownMenuContent>
      </OkuDropdownMenuPortal>
    </OkuDropdownMenu>
  </div>
</template>
