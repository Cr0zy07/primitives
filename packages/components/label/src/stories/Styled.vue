<script setup lang="ts">
import { OkuLabel } from '@oku-ui/label'
</script>

<template>
  <OkuLabel class="label" @click="console.log">
    Label
  </OkuLabel>
</template>
