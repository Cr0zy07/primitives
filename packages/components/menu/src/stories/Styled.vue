<script setup lang="ts">
import { OkuMenuItem, OkuMenuSeparator } from '@oku-ui/menu'
import MenuWithAnchor from './MenuWithAnchor.vue'

function alert(text: string) {
  // eslint-disable-next-line no-alert
  window.alert(text)
}
</script>

<template>
  <MenuWithAnchor>
    <OkuMenuItem class="item" @select="alert('undo')">
      Undo
    </OkuMenuItem>
    <OkuMenuItem class="item" @select="alert('redo')">
      Redo
    </OkuMenuItem>
    <OkuMenuSeparator class="separator" />
    <OkuMenuItem class="item" disabled @select="alert('cut')">
      Cut
    </OkuMenuItem>
    <OkuMenuItem class="item" @select="alert('copy')">
      Copy
    </OkuMenuItem>
    <OkuMenuItem class="item" @select="alert('paste')">
      Paste
    </OkuMenuItem>
  </MenuWithAnchor>
</template>
