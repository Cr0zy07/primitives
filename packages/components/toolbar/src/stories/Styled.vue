<script setup lang="ts">
import ToolbarExample from './ToolbarExample.vue'
</script>

<template>
  <ToolbarExample title="Horizontal" />
  <ToolbarExample title="Vertical" orientation="vertical" />
</template>
