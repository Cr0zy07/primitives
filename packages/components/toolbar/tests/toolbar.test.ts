import { defineComponent } from 'vue'
import { enableAutoUnmount, mount, shallowMount } from '@vue/test-utils'
import type { VueWrapper } from '@vue/test-utils'
import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
import { axe } from 'vitest-axe'

import {
  OkuToolbar,
  OkuToolbarButton,
  OkuToolbarLink,
  OkuToolbarSeparator,
  OkuToolbarToggleGroup,
  OkuToolbarToggleItem,
} from '../src'

import Styled from '../src/stories/Styled.vue'
import Chromatic from '../src/stories/Chromatic.vue'

enableAutoUnmount(afterEach)

const onClick = vi.fn()

const ToolbarTest = defineComponent({
  components: {
    OkuToolbar,
    OkuToolbarToggleGroup,
    OkuToolbarToggleItem,
  },
  setup() {
    return {
      onClick,
    }
  },
  template: `
    <OkuToolbar>
      <OkuToolbarToggleGroup type="single">
        <OkuToolbarToggleItem value="left" @click="onClick">
          Left
        </OkuToolbarToggleItem>
      </OkuToolbarToggleGroup>
    </OkuToolbar>
  `,
})

describe('okuToolbar', () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = mount(ToolbarTest, {
      attachTo: document.body,
    })
  })

  it('should render OkuToolbar correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()

    expect(shallowMount(OkuToolbar).html()).toMatchSnapshot()
  })

  it('should render OkuToolbarToggleItem correctly', () => {
    expect(shallowMount(OkuToolbarToggleItem).html()).toMatchSnapshot()
  })

  it('should render OkuToolbarButton correctly', () => {
    expect(shallowMount(OkuToolbarButton).html()).toMatchSnapshot()
  })

  it('should render OkuToolbarLink correctly', () => {
    expect(shallowMount(OkuToolbarLink).html()).toMatchSnapshot()
  })

  it('should log a warning when the type prop is not provided', () => {
    const spy = vi.spyOn(globalThis.console, 'warn').mockImplementation(() => { })
    const wrapper = () => shallowMount(OkuToolbarToggleGroup)

    expect(() => wrapper()).toThrowErrorMatchingSnapshot()

    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: Missing required prop: "type"')
  })

  it('should render OkuToolbarSeparator correctly', () => {
    const spy = vi.spyOn(globalThis.console, 'warn').mockImplementation(() => { })
    const wrapper = () => shallowMount(OkuToolbarSeparator)

    expect(() => wrapper()).toThrowErrorMatchingSnapshot()

    expect(spy).toHaveBeenCalled()

    expect(spy.mock.calls[0][0]).toContain('[Vue warn]: injection "Symbol(OkuToolbar)" not found.')
  })

  /**
   * @vitest-environment jsdom
   */
  it('should have no accessibility violations', async () => {
    expect(await axe(wrapper.element)).toHaveNoViolations()
  })

  describe('given a default Toolbar', () => {
    beforeEach(() => {
      wrapper = mount(ToolbarTest, {
        attachTo: document.body,
      })
    })

    it('click event should be called just once', async () => {
      const button = wrapper.findAll('button').filter(button => button.text() === 'Left').at(0)!

      await button.trigger('click')

      expect(onClick).toHaveBeenCalledTimes(1)
    })
  })
})

describe('okuToolbar Stories', () => {
  describe('styled', () => {
    let wrapper: VueWrapper<InstanceType<typeof Styled>>

    beforeEach(async () => {
      wrapper = shallowMount(Styled, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('chromatic', () => {
    let wrapper: VueWrapper<InstanceType<typeof Chromatic>>

    beforeEach(async () => {
      wrapper = shallowMount(Chromatic, {
        attachTo: document.body,
      })
    })

    /**
     * @vitest-environment jsdom
     */
    it('should have no accessibility violations', async () => {
      expect(await axe(wrapper.element)).toHaveNoViolations()
    })

    it('should render correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
