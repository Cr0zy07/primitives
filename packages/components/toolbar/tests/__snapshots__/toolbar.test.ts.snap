// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`okuToolbar > should log a warning when the type prop is not provided 1`] = `"\`OkuToolbarToggleGroup\` must be used within \`OkuToolbar\`"`;

exports[`okuToolbar > should render OkuToolbar correctly 1`] = `
"<div role=\\"toolbar\\" ariaorientation=\\"horizontal\\" dir=\\"ltr\\" tabindex=\\"-1\\" data-orientation=\\"horizontal\\" style=\\"outline: none;\\">
  <div dataorientation=\\"horizontal\\" role=\\"group\\" dir=\\"ltr\\" tabindex=\\"0\\" style=\\"outline: none;\\"><button type=\\"button\\" data-state=\\"off\\" role=\\"radio\\" aria-checked=\\"false\\" tabindex=\\"-1\\" data-oku-collection-item=\\"\\"> Left </button></div>
</div>"
`;

exports[`okuToolbar > should render OkuToolbar correctly 2`] = `"<oku-roving-focus-group-stub scopeokurovingfocusgroup=\\"[object Object]\\" orientation=\\"horizontal\\" dir=\\"ltr\\" loop=\\"true\\" aschild=\\"true\\"></oku-roving-focus-group-stub>"`;

exports[`okuToolbar > should render OkuToolbarButton correctly 1`] = `"<oku-roving-focus-group-item-stub scopeokurovingfocusgroup=\\"[object Object]\\" focusable=\\"true\\" active=\\"false\\" aschild=\\"true\\"></oku-roving-focus-group-item-stub>"`;

exports[`okuToolbar > should render OkuToolbarLink correctly 1`] = `"<oku-roving-focus-group-item-stub scopeokurovingfocusgroup=\\"[object Object]\\" focusable=\\"true\\" active=\\"false\\" aschild=\\"true\\"></oku-roving-focus-group-item-stub>"`;

exports[`okuToolbar > should render OkuToolbarSeparator correctly 1`] = `"\`OkuToolbarSeparator\` must be used within \`OkuToolbar\`"`;

exports[`okuToolbar > should render OkuToolbarToggleItem correctly 1`] = `"<oku-toolbar-button-stub disabled=\\"false\\" aschild=\\"true\\"></oku-toolbar-button-stub>"`;

exports[`okuToolbar Stories > chromatic > should render correctly 1`] = `
"<div style=\\"padding: 50px;\\">
  <h1>Example</h1>
  <toolbar-example-stub></toolbar-example-stub>
  <toolbar-example-stub orientation=\\"vertical\\"></toolbar-example-stub>
  <h1>Direction</h1>
  <h2>Prop</h2>
  <toolbar-example-stub dir=\\"rtl\\"></toolbar-example-stub>
  <h2>Inherited</h2>
  <direction-provider-stub dir=\\"rtl\\"></direction-provider-stub>
</div>"
`;

exports[`okuToolbar Stories > styled > should render correctly 1`] = `
"<toolbar-example-stub title=\\"Horizontal\\"></toolbar-example-stub>
<toolbar-example-stub title=\\"Vertical\\" orientation=\\"vertical\\"></toolbar-example-stub>"
`;
